lib/spree/shipment_notice.rb

```diff
@@ -27,14 +27,20 @@ def locate
     # => true
     def update
       shipment.update_attribute(:tracking, tracking)
+      order = shipment.order
+      result = !order.outstanding_balance?
+
+      if Spree::Config.shipstation_capture_at_notification && !result
+        result = order.capture_payments!
+      end
 
       unless shipment.shipped?
         shipment.reload.ship!
         shipment.touch :shipped_at
         shipment.order.update!
       end
 
-      true
+      result
```

Comment:
I changed the logic first we locate the shipment, if not found we return false and set errors,
if found but not paid then we return false and set errors we only update tracking number and
change shipment state if found and paid.